### DISCUSSION TITLE
想請問可不可以使用量化過後的中文模型(GGUF格式)來訓練

### DISCUSSION THREAD

**Discussion by james60415** [ymcui/Chinese-LLaMA-Alpaca-2#471] [General]
目前使用v100 16G顯示卡按照wiki的[預訓練腳本](https://github.com/ymcui/Chinese-LLaMA-Alpaca-2/wiki/pt_scripts_zh)訓練7B的中文模型會OOM，所以想改使用8bit或6bit量化的GGUF模型來訓練。想問可以這樣訓練嗎，如果可以的訓練的話訓練步驟按照[llama.cpp的手冊](https://github.com/ggerganov/llama.cpp/tree/master/examples/finetune)做可以嗎

**Comment by ymcui** [comment]
GGUF是llama.cpp独有格式，与训练脚本（基于transformers、PyTorch）不兼容。

**Reply by james60415** [reply]
感謝您的回答與補充，那我把中文量化的模型用llama.cpp的訓練步驟是可行的嗎

**Reply by ymcui** [reply]
llama.cpp主要是做量化推理的，不是用来训练和精调大模型的。
llama.cpp提供了一个叫`train-text-from-scratch`的程序，可以做一些简单的训练，但具体用法恐怕需要你自己去llama.cpp那边查找（并不是一个常用的功能）。

**Reply by james60415** [reply]
非常感謝您撥空回覆，我會再自行研究`train-text-from-scratch`，不過llama cpp也有推出`finetune`的程序([連結](https://github.com/ggerganov/llama.cpp/tree/master/examples/finetune))，應該也是可以做訓練的，這部分我也會再自行研究，謝謝您
